sbc_compassion/models/import_letter_line.py

```diff
@@ -78,10 +79,13 @@ def _set_sponsorship_id(self):
                     [('child_id.code', '=', inst.child_code)]) and
                     inst.env['recurring.contract'].search([
                         ('partner_codega', '=', inst.partner_codega)]))
-
-                inst.sponsorship_id = inst.env['recurring.contract'].search([
+                temp = inst.env['recurring.contract'].search([
```

Comment:
Avoid names like `temp`. You could directly name it `sponsorship` and instead of having the function `_check_sponsorship`, you can modify your search like this: 
```
inst.env['recurring.contract'].search([
    ('child_code', '=', inst.child_code),
    ('partner_codega', '=', inst.partner_codega),
    ('is_active', '=', True)], order='end_date desc', limit=1)